From 09077caa8202cba5df0578fb847280d3dd721317 Mon Sep 17 00:00:00 2001
From: Alicia Klinvex <amklinv@sandia.gov>
Date: Thu, 23 Mar 2017 07:53:14 -0600
Subject: [PATCH] Modified CMakeLists to work with spack

Spack does not define PETSC_ARCH, so the configuration fails at
line 191 when PETSC_ARCH is compared to "".  This modification
causes alquimia to ignore PETSC_ARCH if it is not defined.
---
 CMakeLists.txt | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b17540c..f00e027 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,11 +188,10 @@ if (ALQUIMIA_NEED_PETSC)
   if ($ENV{PETSC_DIR} STREQUAL "")
     message(FATAL_ERROR "PETSC_DIR must be set for the requested engines to be enabled.")
   endif()
-  if ($ENV{PETSC_ARCH} STREQUAL "")
-    message(FATAL_ERROR "PETSC_ARCH must be set for the requested engines to be enabled.")
-  endif()
   set(PETSC_DIR $ENV{PETSC_DIR})
-  set(PETSC_ARCH $ENV{PETSC_ARCH})
+  if(DEFINED $ENV{PETSC_ARCH})
+    set(PETSC_ARCH $ENV{PETSC_ARCH})
+  endif()
 
   message(STATUS "--- SM: PETSC_DIR >>> ${PETSC_DIR}")
   message(STATUS "--- SM: PETSC_ARCH >>> ${PETSC_ARCH}")
@@ -234,7 +233,11 @@ if (ALQUIMIA_NEED_PETSC)
   include_directories(${PETSC_INCLUDES})
 
   # Include PETSc in the rpath.
-  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${PETSC_DIR}/${PETSC_ARCH}/lib")
+  if(DEFINED ${PETSC_ARCH})
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${PETSC_DIR}/${PETSC_ARCH}/lib")
+  else()
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${PETSC_DIR}/lib")
+  endif()
   message(STATUS "--- SM: cmake install rpath")
 endif()
 

